### DISCUSSION TITLE
PIP Dependency ERROR

### DISCUSSION THREAD

**Discussion by glovas1** [beeware/toga#1533] [Q&A]
I have attached the output I get when trying to run 'briefcase create android' for an app I built which has been working fine up to this point. Now this error seems to indicate something is out of whack with the versions but I am not sure if there is something I can do about it. The error message is : 

ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
toga-winforms 0.3.0.dev33 requires toga-core==0.3.0.dev33, but you have toga-core 0.3.0.dev35 which is incompatible.

Any guidance would be very much appreciated!
[pip_error.txt](https://github.com/beeware/toga/files/9193931/pip_error.txt)

Sorry, this should have been under Discussions

**Comment by freakboy3742** [comment]
This indicates that you've got an inconsistency between your development environment and your `pyproject.toml` declarations - as the error says, your development environment has dev33, but your pyproject specifies dev35 (or something that will resolve to dev35, like `>= dev30`).

However - as you may have noted, the app is completing installation, and it's reporting that dev35 of toga-core and toga-android have both been installed. The error is a bit of a misnomer, caused by the fact that we're using the `pip` install in your development environment to install the Android packages. This causes a bit of confusion for the pipe dependency resolver which it *reports* as an error, even though there isn't actually a problem in practice.

**Reply by glovas1** [reply]
Thanks for the quick answer and direction.    :-)

**Reply by mhsmith** [reply]
See also https://github.com/beeware/briefcase/issues/768.
